aptos-node/src/storage.rs

```diff
@@ -1,44 +1,74 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-use anyhow::anyhow;
+use anyhow::{anyhow, Result};
+use aptos_backup_service::start_backup_service_with_fast_sync_wrapper;
 use aptos_config::{config::NodeConfig, utils::get_genesis_txn};
-use aptos_db::AptosDB;
+use aptos_db::{fast_sync_aptos_db::FastSyncStorageWrapper, AptosDB};
 use aptos_executor::db_bootstrapper::maybe_bootstrap;
 use aptos_logger::{debug, info};
-use aptos_storage_interface::{DbReader, DbReaderWriter};
+use aptos_storage_interface::{DbReader, DbReaderWriter, FastSyncStatus};
 use aptos_types::waypoint::Waypoint;
 use aptos_vm::AptosVM;
-use std::{fs, net::SocketAddr, path::Path, sync::Arc, time::Instant};
+use std::{
+    fs,
+    path::Path,
+    sync::{Arc, RwLock},
+    time::Instant,
+};
 use tokio::runtime::Runtime;
 
 #[cfg(not(feature = "consensus-only-perf-test"))]
 pub(crate) fn bootstrap_db(
-    aptos_db: AptosDB,
-    backup_service_address: SocketAddr,
-) -> (Arc<AptosDB>, DbReaderWriter, Option<Runtime>) {
-    use aptos_backup_service::start_backup_service;
+    node_config: &NodeConfig,
+) -> Result<(Arc<FastSyncStorageWrapper>, DbReaderWriter, Option<Runtime>)> {
+    let fast_sync_status = Arc::new(RwLock::new(FastSyncStatus::UNKNOWN));
+    let db = FastSyncStorageWrapper::new_fast_sync_aptos_db(node_config, fast_sync_status.clone())?;
```

Comment:
I also thought of this but didn't find a good way.  eventually, I think it is better to have all the logic inside FastSyncStorageWrapper instead of exposing the complexity of selecting different DB inside aptos-node. Eg: the fast sync storage wrapper will decide not to create the secondary DB if the config is not turned on. But happy to know if there is a better way to do this.

Reply:
I also thought of this but didn't find a good way.  eventually, I think it is better to have all the logic inside FastSyncStorageWrapper instead of exposing the complexity of selecting different DB inside aptos-node. Eg: the fast sync storage wrapper will decide not to create the secondary DB if the config is not turned on. But happy to know if there is a better way to do this.

Reply:
updated based on offline discussion